docs/digger/installation/macos-prereqs.asciidoc

```diff
@@ -1,41 +1,132 @@
+
+dscl . create /Users/${USERNAME} UniqueID ${USER_ID}
+dscl . create /Users/${USERNAME} PrimaryGroupID ${GROUP_ID}
+dscl . create /Users/${USERNAME} UserShell /bin/bash
+dscl . create /Users/${USERNAME} NFSHomeDirectory /Users/${USERNAME}
+dseditgroup -o edit -a ${USERNAME} -t user admin
+cp -R /System/Library/User\ Template/English.lproj /Users/${USERNAME}
+chown -R ${USERNAME}:${GROUP_NAME} /Users/${USERNAME}
+
+echo "${USERNAME}  ALL=(ALL:ALL) ALL" >> /etc/sudoers
+
+echo "Done creating OSX user - you may need to restart the osx server to apply all changes for the user ${USERNAME}"
 ----
-+
-. Execute `enable-osx-remote-login.sh` to allow SSH access:
-+
+
+By default, this script will create a user named `jenkins` with SSH access. If using the above script you should read it carefully
+and understand its implications. The `PASSWORD` variable should be changed to something secure.
+
+
+This script, or individual commands to achieve the same state will need to be run on the macOS server. It can be downloaded for
+convenience using the following command: 
+
+`curl -O https://raw.githubusercontent.com/aerogear/digger-jenkins/master/admin/create-osx-user.sh`
+
+Restart the server to implement the changes made above.
+  
+====== Configure Remote Login
+
+We must enable the Ansible process running remotely to execute commands on the macOS server via SSH.
+
+Again, a sample script is provided to achieve this, `USERNAME` variable should match the user that was created in the previous step.
+
+[source,bash]
 ----
-sudo bash enable-osx-remote-login.sh
+#!/usr/bin/env bash
+
+# This script helps with enabling SSH for given OSX user.
+# This script is not meant to be run in an automation. Run it manually.
+
+USERNAME="jenkins"
+
+# com.apple.access_ssh is a special group name on OSX.
+# any user part of that group can have SSH connections in.
+OSX_SSH_GROUP_NAME="com.apple.access_ssh"
+
+systemsetup -setremotelogin on
+# in order to check what groups are are there:
+#   dscl . list /Groups PrimaryGroupID
+# create a group for limiting SSH access
+dseditgroup -o create -q ${OSX_SSH_GROUP_NAME}
+# add user into this group
+dseditgroup -o edit -a ${USERNAME} -t user ${OSX_SSH_GROUP_NAME}
+# now, following should work  ---> ssh username@localhost
 ----
---
+
+Again this can be downloaded by running the following:
+
+`curl -O https://raw.githubusercontent.com/aerogear/digger-jenkins/master/admin/enable-osx-remote-login.sh`
+
+
+====== Connecting via Ansible
+
+A number of tasks in the `provision-osx` Ansible role used in the installer require http://docs.ansible.com/ansible/latest/become.html[privilege escalation]. 
+This allows us to run commands as `sudo` user
+
+This user created in the previous steps should be provided as a host variable to Ansible via `ansible_ssh_user`
+
+Set `ansible_sudo_pass` variable or pass the `--ask-sudo-pass` flag when running the Ansible playbook to enable the installer to work with root permissions.
```

Comment:
Should we replace it with "ansible_become_pass" and "--ask-become-pass"?

Reply:
We have a Jira to replace also the rhmap-installer ones but for now I think we should leave as is for consistency and address them all in the same JIRA for next release. Does that sound okay?

Reply:
👍